### ISSUE TITLE
error, how to figure out, I can't run this demo

### ISSUE THREAD

**Issue by shuzizihemu** [lcy0604/EraseNet#11]
![image](https://user-images.githubusercontent.com/48058763/197393627-3d1cd1a0-6e84-4037-b4e9-fb0a48681cee.png)

**Comment by TD-wzw** [comment]
报错原因是因为在Loss.py中的self.D_optimizer.step()这一句的存在
可以把self.D_optimizer传出去，在G_loss.backward()之后再去更新

**Comment by Aspen0309** [comment]
> 报错原因是因为在Loss.py中的self.D_optimizer.step()这一句的存在 可以把self.D_optimizer传出去，在G_loss.backward()之后再去更新

您好，我传出去之后输出结果都是黑图，可以贴下你的代码嘛

**Comment by shallweiwei** [comment]
I can share a new way here.You can use detach when calculate dfake for D_loss().And after D_loss had backward,then calculate another dfake using discriminator for G_loss().

**Comment by ray7jq** [comment]
您好，您能贴出你的代码吗？

**Comment by seniordba2023** [comment]
![Image](https://github.com/user-attachments/assets/fe9f82cb-614e-46e5-9db1-53cc602a8047)

![Image](https://github.com/user-attachments/assets/0c326337-e73b-4a42-8502-a71ad408e1e2)
根据作者得指导，我这样调整后，已经跑通了
